src/libcprover-cpp/api.cpp

```diff
@@ -0,0 +1,54 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#include "api.h"
+
+#include <util/cmdline.h>
+#include <util/config.h>
+#include <util/message.h>
+#include <util/options.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/initialize_goto_model.h>
+
+#include <ansi-c/ansi_c_language.h>
+#include <langapi/mode.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+extern configt config;
+
+api_sessiont::api_sessiont()
+  : message_handler(
+      util_make_unique<null_message_handlert>(null_message_handlert{})),
+    options(util_make_unique<optionst>(optionst{}))
+{
+  // Needed to initialise the language options correctly
+  cmdlinet cmdline;
+  // config is global in config.cpp
+  config.set(cmdline);
+  // Initialise C language mode
+  register_language(new_ansi_c_language);
+}
+
+api_sessiont::api_sessiont(const api_optionst &options)
+  : message_handler(
+      util_make_unique<null_message_handlert>(null_message_handlert{})),
```

Comment:
Enforcing a null message handler hardly seems like a good idea. If you absolutely want to avoid exposing the `message_handlert` class, then just use a `stream_message_handlert` and require the caller to provide a `std::ostream`.